agent/src/beerocks/slave/son_slave_thread.cpp

```diff
@@ -4315,8 +4315,32 @@ bool slave_thread::handle_autoconfiguration_wsc(Socket *sd, ieee1905_1::CmduMess
     }
 
     for (auto tlvWscM2 : m2_list) {
+        auto encrypted_settings = tlvWscM2->encrypted_settings();
+        auto ssid  = std::string(encrypted_settings->ssid(), encrypted_settings->ssid_length());
+        auto bssid = network_utils::mac_to_string(encrypted_settings->bssid_attr().data);
+        auto authentication_type = encrypted_settings->authentication_type_attr().data;
+        auto encryption_type     = encrypted_settings->encryption_type_attr().data;
+        // char network_key[WSC::WSC_MAX_NETWORK_KEY_LENGTH];
+        // std::copy_n(encrypted_settings.network_key_attr().data,
+        //             encrypted_settings.network_key_attr().data_length,
+        //             network_key);
+        // char key_wrap_auth[WSC::WSC_KEY_WRAP_AUTH_LENGTH];
+        // std::copy_n(encrypted_settings.key_wrap_auth_attr().data,
+        //             encrypted_settings.key_wrap_auth_attr().data_length,
+        //             network_key);
         std::string manufacturer =
             std::string(tlvWscM2->manufacturer(), tlvWscM2->manufacturer_length());
+        // ONLY FOR DEBUG!!!
+        // TODO - remove this print, it is showing the encrypted settings in the logs!
+        // TODO - looks like authentication_type and encryption_type are not swapped.
```

Comment:
Just a thought - you didn't finalize yet, why do you expect them to be swapped?

Reply:
On RX, when you call parse, it does the swapping.

Reply:
I need to remove the TODO - they are swapped, this was a bug fixed in other commit